pkg/services/secrets/kvstore/remote_plugin_check.go

```diff
@@ -1,26 +1,30 @@
 package kvstore
 
 import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/plugins/backendplugin/secretsmanagerplugin"
 )
 
 type UseRemoteSecretsPluginCheck interface {
 	ShouldUseRemoteSecretsPlugin() bool
-	GetPlugin() (secretsmanagerplugin.SecretsManagerPlugin, error)
+	StartAndReturnPlugin(ctx context.Context) (secretsmanagerplugin.SecretsManagerPlugin, error)
 }
 
-type OSSRemoteSecretsPluginCheck struct {
-	UseRemoteSecretsPluginCheck
-}
+type OSSRemoteSecretsPluginCheck struct{}
 
 func ProvideRemotePluginCheck() *OSSRemoteSecretsPluginCheck {
 	return &OSSRemoteSecretsPluginCheck{}
 }
 
-func (c *OSSRemoteSecretsPluginCheck) ShouldUseRemoteSecretsPlugin() bool {
+func (c OSSRemoteSecretsPluginCheck) ShouldUseRemoteSecretsPlugin() bool {
 	return false
 }
 
-func (c *OSSRemoteSecretsPluginCheck) GetPlugin() (secretsmanagerplugin.SecretsManagerPlugin, error) {
+func (c OSSRemoteSecretsPluginCheck) StartAndReturnPlugin(ctx context.Context) (secretsmanagerplugin.SecretsManagerPlugin, error) {
+	log.New("ossremotesecretsplugincheck").Error("OSSRemoteSecretsPluginCheck.StartAndReturnPlugin() should not have been called")
```

Comment:
Maybe we could make this error log friendlier? Maybe something like "secrets manager plugins are enterprise only". I think it's more importat to say why it shouldn't have been called.

Reply:
done